app/assets/stylesheets/new_styles/_forms.scss

```diff
@@ -4,122 +4,102 @@ input:-webkit-autofill{
   background-image: none !important;
 }
 
-
 form.block-form {
-  display : inline-block;
+  margin: 20px auto;
 
   label {
-    color : $text-dark-grey;
-  }
-
-  input {
-    &:invalid,
-    &:invalid:required,
-    &:invalid:required:focus {
-      -webkit-box-shadow : none;
-      -moz-box-shadow : none;
-      box-shadow : none;
-      box-shadow : none;
-
-      border : none;
-      color : $text-dark-grey;
-    }
+    color: $text-dark-grey;
+    position: absolute;
+    top: -9999px;
+    left: -9999px;
   }
 
   fieldset {
-    @include border-radius(5px);
-
-    -webkit-box-shadow : inset 0 1px 1px rgba(0,0,0,0.2), 0 1px 2px rgba(255,255,255,0.7);
-
-    margin-bottom : 1em;
-
-    background-color : #fff;
-
-    border : 1px solid $border-dark-grey;
-
-    input[type=text],
-    input[type=email],
-    input[type=password] {
-      @include box-shadow(0,0,0,0);
-      @include border-radius(0);
-
-      background : transparent;
-
-      padding : 10px;
-
-      margin-bottom : 0;
-      border : none;
+    margin-bottom: 1em;
+    background-color: #fff;
+    position: relative; /* To correctly place the entypo icon */
+  
+    /* Bootstrap reset */
+    input {
+      &,
+      &:focus, 
+      &:invalid,
+      &:invalid:focus,
+      &:invalid:required,
+      &:invalid:required:focus {
+        border: 1px solid $border-grey;
+        border-bottom: 0px;
+        box-shadow: none;
+        color : $text-dark-grey;
+        margin: 0px;
+      }
     }
 
-  /* mainly bootstrap overrides */
-    .control-group {
-      margin : 0;
-      border-bottom : 1px solid $border-grey;
-
-      .control-label,
-      input[type=text],
-      input[type=password],
-      .field_with_errors label {
-        padding : 10px;
-        margin : 0;
+    .form-control {
+      font-size: 16px;
+      height: 40px;
+      padding: 10px;
+      padding-left: 40px;
+    }
+    
+    .form-control:first-of-type {
+      @include border-radius(5px, 5px, inherit, inherit);
+    }
+    
+    .form-control:last-of-type {
+      &,
+      &:focus, 
+      &:invalid,
+      &:invalid:focus,
+      &:invalid:required,
+      &:invalid:required:focus {
```

Comment:
Since we are already using boostrap-sass, maybe tune-up to bootstrap can be done by using Sass variables? You can customize almost the whole framework there, and it will look more consistent, because you can't forget to apply the styling to some element.